04 - Poker.playground/Contents.swift

```diff
@@ -6,17 +6,290 @@
+                if cards[i].rank == cards[j].rank {
+                    sameRank += 1
+                }
+            }
+            switch cards[i].rank {
+            case .deuce:
+                cardRanks[1] += 1
+            case .three:
+                cardRanks[2] += 1
+            case .four:
+                cardRanks[3] += 1
+            case .five:
+                cardRanks[4] += 1
+            case .six:
+                cardRanks[5] += 1
+            case .seven:
+                cardRanks[6] += 1
+            case .eight:
+                cardRanks[7] += 1
+            case .nine:
+                cardRanks[8] += 1
+            case .ten:
+                cardRanks[9] += 1
+            case .jack:
+                cardRanks[10] += 1
+            case .queen:
+                cardRanks[11] += 1
+            case .king:
+                cardRanks[12] += 1
+            case .ace:
+                cardRanks[13] += 1
+                cardRanks[0] += 1
+            }
+        }
+        
+        if sameRank == 7 {
+            return Ranking.Pair
+        } else if sameRank == 9 {
+            return Ranking.TwoPairs
+        } else if sameRank == 11 {
+            return Ranking.ThreeOfAKind
+        } else if sameRank == 13 {
+            return Ranking.FullHouse
+        } else if sameRank == 17 {
+            return Ranking.FourOfAKind
+        } else if cards[0].suit == cards[1].suit && cards[1].suit == cards[2].suit && cards[2].suit == cards[3].suit && cards[3].suit == cards[4].suit {
+            if cardRanks == [1,1,1,1,1,0,0,0,0,0,0,0,0,0] || cardRanks == [0,1,1,1,1,1,0,0,0,0,0,0,0,0] || cardRanks == [0,0,1,1,1,1,1,0,0,0,0,0,0,0] || cardRanks == [0,0,0,1,1,1,1,1,0,0,0,0,0,0] || cardRanks == [0,0,0,0,1,1,1,1,1,0,0,0,0,0] || cardRanks == [0,0,0,0,0,1,1,1,1,1,0,0,0,0] || cardRanks == [0,0,0,0,0,1,1,1,1,1,0,0,0,0] || cardRanks == [0,0,0,0,0,0,1,1,1,1,1,0,0,0] || cardRanks == [0,0,0,0,0,0,0,1,1,1,1,1,0,0] || cardRanks == [0,0,0,0,0,0,0,0,1,1,1,1,1,0] || cardRanks == [0,0,0,0,0,0,0,0,0,1,1,1,1,1] {
+                print("Awesome! Straight flush: " + cards[0].Description + cards[1].Description + cards[2].Description + cards[3].Description + cards[4].Description)
+                return Ranking.StraightFlush
+            } else {
+                return Ranking.Flush
+            }
+        } else if cardRanks == [1,1,1,1,1,0,0,0,0,0,0,0,0,0] || cardRanks == [0,1,1,1,1,1,0,0,0,0,0,0,0,0] || cardRanks == [0,0,1,1,1,1,1,0,0,0,0,0,0,0] || cardRanks == [0,0,0,1,1,1,1,1,0,0,0,0,0,0] || cardRanks == [0,0,0,0,1,1,1,1,1,0,0,0,0,0] || cardRanks == [0,0,0,0,0,1,1,1,1,1,0,0,0,0] || cardRanks == [0,0,0,0,0,1,1,1,1,1,0,0,0,0] || cardRanks == [0,0,0,0,0,0,1,1,1,1,1,0,0,0] || cardRanks == [0,0,0,0,0,0,0,1,1,1,1,1,0,0] || cardRanks == [0,0,0,0,0,0,0,0,1,1,1,1,1,0] || cardRanks == [0,0,0,0,0,0,0,0,0,1,1,1,1,1] {
+            return Ranking.Straight
+        } else {
+            return Ranking.OtherStuff
+        }
+    }
```

Comment:
Funktioniert offenbar ^^ Ist praktischer einzelne Tests in Funktionen oder Computed Properties auszulagern wie z.B. `pairCount: Int`, `hasFlush: Bool`, `hasStraight: Bool` usw., dann ist das alles übersichtlicher.
Aber solange das Ergebnis stimmt passts 😉